File: app/src/main/java/com/task/paginationlist/presentation/piclist/presenter/BasePresenter.java
package com.task.paginationlist.presentation.piclist.presenter;

import android.support.annotation.NonNull;

import com.arellomobile.mvp.MvpPresenter;
import com.arellomobile.mvp.MvpView;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class BasePresenter<V extends MvpView> extends MvpPresenter<V> {

    private CompositeDisposable compositeSubscription = new CompositeDisposable();

    protected void addToCompositeSubscription(@NonNull Disposable subscription) {
        compositeSubscription.add(subscription);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        compositeSubscription.clear();
    }
}

File: app/src/main/java/com/task/paginationlist/presentation/piclist/presenter/BasePresenter.kt
package com.task.paginationlist.presentation.piclist.presenter

import com.arellomobile.mvp.MvpPresenter
import com.arellomobile.mvp.MvpView

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

open class BasePresenter<V : MvpView> : MvpPresenter<V>() {

    private val compositeSubscription = CompositeDisposable()

    protected fun addToCompositeSubscription(subscription: Disposable) {
        compositeSubscription.add(subscription)
    }

    override fun onDestroy() {
        super.onDestroy()
        compositeSubscription.clear()
    }
}
